fluent/ChainRune.ts

```diff
@@ -50,9 +51,41 @@ export namespace Chain {
 
 // TODO: do we want to represent the discovery value and conn type within the type system?
 export class ChainRune<out C extends Chain, out U> extends Rune<C, U> {
+  static from<X, M extends FrameMetadata>(
+    props: RunicArgs<X, { connection: Connection; metadata?: M }>,
+  ) {
+    return Rune
+      .rec(props)
+      .unsafeAs<{ connection: Connection; metadata: M }>()
+      .into(ChainRune)
+  }
+
+  static dynamic<D>(
+    connectionCtor: {
```

Comment:
Simpler way to represent this type? The following won't work

- `typeof Connection` –– type compat error
- `new(discovery: D) => Connection` –– we need access to the `connect` static
- `{ connect: (discovery: D, signal: AbortSignal) => Connection }` –– for some reason infers `D` as unknown